build_tools/ci/cpu_comparison/run.py

```diff
@@ -37,35 +38,48 @@ def run_conv_test(config, filename, n_repeats):
     return True
 
 
-class BaseTemplate(ABC):
+class BaseTest(ABC):
     """
-    Base class to be inherited by any new dispatch/compute op template.
-    The derived instances would therefore be created specifying the intended target
-    device(s) they're to be run; and would accordingly be `run` if the intended
-    target device(s) contains the `target_device` supplied via command-line.
+    Base class to be inherited by all tests. The derived instances will be
+    created specifying the intended target device(s) they're to be run on; and
+    will accordingly be `run` if the intended target device(s) contains
+    the `target_device` found in `config`. The default set of targets to run on
+    is the singleton set `["npu1_4col"]`.
+
+    An instance of this class has a member `aie_compilation_flags`
+    which are additional flags to be passed to the AIE backend compiler.
+
+    Compilation flags can therefore be injected in 2 ways:
+    1) via the constructor of this base class
+    2) via the `add_aie_compilation_flags` method
     """
 
-    def __init__(self, run_on_target, additional_aie_compilation_flags=None):
+    def __init__(self, run_on_target=["npu1_4col"], aie_compilation_flags=[]):
         self.run_on_target = run_on_target
-        self.additional_aie_compilation_flags = additional_aie_compilation_flags \
-            if additional_aie_compilation_flags is not None else []
+        self.aie_compilation_flags = aie_compilation_flags
+        # NB: derived classes should add labels to this list in their
+        # constructor, never overwrite it.
+        self.labels = ["All"]
```

Comment:
What is "All" used for in the base class? Why it's not empty?

Reply:
If tests are run with for example

```
./run.py ...   --target_device=npu1_4col --xrt_lite_n_core_rows=4 --xrt_lite_n_core_cols=4 --tests=VanillaMatmul
```

then all the tests which have label `VanillaMatmul` will be run. 

All the tests inherit from this base class, and so they all have the label `All`. 

So if you run 

```
./run.py ...   --target_device=npu1_4col --xrt_lite_n_core_rows=4 --xrt_lite_n_core_cols=4 --tests=VanillaMatmul
```
then all the tests will run. Well, there is additional check for tests which use microkernel that chess is available, but that's the basic idea

Reply:
(tests can have multiple labels)

Reply:
What is not clear to me is all other classes will append to the labels from the baseClass. So if I create an instance of VanillaMatmul class, it will run "All" and "VanillaMatmul"?

Reply:
If you create an instance of VanillaMatmul, it will have labels "All" and "VanillaMatmul". It will be run if the user specifies `tests=All` or `tests=VanillaMatmul`. At least that's what I've designed it to do.

Reply:
So if I understand correctly, if I specify `tests=VanillaMatmul`, it will only run those tests within VanillaMatmul class, but it has the `labels = ["All", "VanillaMatmul"]`?

Reply:
https://github.com/nod-ai/iree-amd-aie/blob/4c3bbb37e3dc5173dfd10332c10ef94c663fd97e/build_tools/ci/cpu_comparison/run.py#L1186

> So if I understand correctly, if I specify tests=VanillaMatmul, it will only run those tests within VanillaMatmul class,

Yes

> but it has the labels = ["All", "VanillaMatmul"]

Yes, so all tests will run if you specify `tests=All`